apps/extension/src/core/managers/transaction.ts

```diff
@@ -41,28 +52,49 @@ class TransactionManager extends BaseManager<Transaction> {
   init<TInput extends Input>(
     input: TInput,
     origin: OriginData,
-    options: CompletionOptions<ModelID | string, TInput>
+    options: CompletionOptions<ModelID | string, TInput> | MediaOptions<ModelID>
   ): Transaction {
     this._validateInput(input)
+  
+    // Extracting parameters common to both CompletionOptions and MediaOptions
     const {
-      temperature,
-      maxTokens,
-      stopSequences,
       model,
-      numOutputs = 1
+      numOutputs = 1,
     } = options
+  
+    // Extracting parameters specific to CompletionOptions
+    const {
+      temperature,
+      maxTokens,
+      stopSequences
+    } = options as CompletionOptions<ModelID | string, TInput>
+  
+    // Extracting parameters specific to MediaOptions
+    const {
+      mimeType,
+    } = options as MediaOptions<ModelID | string>
+
+    //extracting parameters specific to 3d generation
```

Comment:
why is this specific to 3d generation? numInferenceSteps is also for 2D, right?

Reply:
it depends, image models like stability do, openai's dalle doesn't allow you to set `numInferenceSteps`, instead just a string `size` property, so I thought we could keep it specific to for 3D for now and when we add images we can decide which properties to include.